### PR TITLE
Clean up use of PROJECT_SOURCE_PATH in tests

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🧹 Clean some uses of PROJECT_SOURCE_PATH

This is partially to address compatibility with Bazel builds, but also generally better practice going forward as it handles paths more consistently across operating systems.

Signed-off-by: Michael Carroll <michael@openrobotics.org>

### CODE REVIEW

File: test/integration/nested_model.cc [hunk 1]
```diff
@@ -1122,7 +1121,7 @@ TEST_F(PlacementFrame, ModelPlacementFrameAttribute)
 TEST(NestedReference, PoseRelativeTo)
 {
   const std::string testFile =
-    sdf::filesystem::append(PROJECT_SOURCE_PATH, "test", "sdf",
+    sdf::testing::TestFile("sdf",
```
Comment (ahcorde) [inline test/integration/nested_model.cc:1124]:
```suggestion
    sdf::testing::TestFile("sdf",
```

File: test/integration/nested_model.cc [hunk 2]
```diff
@@ -1185,7 +1184,7 @@ TEST(NestedReference, PoseRelativeTo)
 TEST(NestedReference, PoseRelativeToInWorld)
 {
   const std::string testFile =
-    sdf::filesystem::append(PROJECT_SOURCE_PATH, "test", "sdf",
+    sdf::testing::TestFile("sdf",
```
Comment (ahcorde) [inline test/integration/nested_model.cc:1187]:
```suggestion
    sdf::testing::TestFile("sdf",
```

File: test/integration/nested_model.cc [hunk 3]
```diff
@@ -1246,7 +1245,7 @@ TEST(NestedReference, PoseRelativeToInWorld)
 TEST(NestedReference, PlacementFrameAttribute)
 {
   const std::string testFile =
-      sdf::filesystem::append(PROJECT_SOURCE_PATH, "test", "integration",
+      sdf::testing::TestFile("integration",
```
Comment (ahcorde) [inline test/integration/nested_model.cc:1248]:
```suggestion
      sdf::testing::TestFile("integration",
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: tools/code_check.sh [hunk 4]
```diff
@@ -11,7 +11,7 @@ if test "$1" = "-xmldir" -a -n "$2"; then
   builddir=../build
 else
   # This is a heuristic guess; not every developer puts the `build` dir in the src dir
-  builddir=./build
+  builddir=/home/mjcarroll/workspaces/ign_edifice/build/sdformat11/
```
Comment (azeey) [inline tools/code_check.sh:14]: This seems unintended.

Comment (mjcarroll) [reply tools/code_check.sh]: Yep, thanks

**Review by azeey — APPROVED** [review]
LGTM!

**Review by ahcorde — APPROVED** [review]

**Pull request merged**